connector/mqtt/vernemq/v2k-bridge/app/Utils.js

```diff
@@ -19,12 +19,37 @@ const generateDojotDeviceDataMessage = (topic, payload) => {
 
   const tenantValue = splitUsername[0];
   const deviceIdValue = splitUsername[1];
+  let metadata = { timestamp: 0 };
+
+  if ("timestamp".indexOf(payload)) {
+    // metadata = { timestamp: 0 };
+    // If it is a number, just copy it. Probably Unix time.
+    if (typeof payload.timestamp === "number") {
+      if (!Number.isNaN(payload.timestamp)) {
+        metadata.timestamp = payload.timestamp;
+      } else {
+        this.logger.info('Received an invalid timestamp (NaN)');
```

Comment:
You need import Logger tool to use here

Reply:
Done, added 'const { Logger } = require('@dojot/microservice-sdk');' following example from vernMQ and pass into CodeFactor.